platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/SimpleListItem.kt

```diff
@@ -12,23 +12,234 @@ import androidx.compose.runtime.Composable
+ * A simple list item layout comprising of a content slot and an optional icon to its start side.
+ *
+ * @param state The state of the list item, containing selection and activity status.
+ * @param colorFilter Optional [ColorFilter] applied to the item, typically used with the icon.
+ * @param painterHints Optional list of [PainterHint] to provide hints for painting customizations; default is empty.
+ * @param modifier Optional [Modifier] to apply to the root container of the list item.
+ * @param iconModifier Optional [Modifier] to apply to the icon displayed in the list item.
+ * @param icon Optional [IconKey] that defines which icon should be displayed.
+ * @param iconContentDescription Optional content description [String] for the icon, used for accessibility.
+ * @param style The style of the list item, including colors and metrics, with a default value from the theme; default
+ *   is based on the Jewel theme.
+ * @param height The height of the list item; default is based on the Jewel theme's global metrics.
+ * @param content A composable lambda representing the main content of the list item.
+ */
+@Composable
+public fun SimpleListItem(
+    state: ListItemState,
+    colorFilter: ColorFilter? = null,
+    painterHints: List<PainterHint> = emptyList(),
+    modifier: Modifier = Modifier,
+    iconModifier: Modifier = Modifier,
+    icon: IconKey? = null,
+    iconContentDescription: String? = null,
+    style: SimpleListItemStyle = JewelTheme.simpleListItemStyle,
+    height: Dp = JewelTheme.globalMetrics.rowHeight,
+    content: @Composable () -> Unit,
+) {
+    Row(
+        modifier =
+            modifier
+                .semantics { selected = state.isSelected }
+                .fillMaxWidth()
+                .height(height)
+                .padding(style.metrics.outerPadding)
+                .background(
+                    color = style.colors.backgroundFor(state).value,
+                    shape = RoundedCornerShape(style.metrics.selectionBackgroundCornerSize),
+                )
+                .padding(style.metrics.innerPadding),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(style.metrics.iconTextGap),
+    ) {
+        if (icon != null) {
+            Icon(
+                modifier = iconModifier.size(16.dp),
+                key = icon,
+                contentDescription = iconContentDescription,
+                colorFilter = colorFilter,
+                hints = painterHints.toTypedArray(),
+            )
+        }
+        content()
+    }
+}
+
 /**
  * A simple list item layout comprising of a content slot and an optional icon to its start side.
  *
  * The text will only take up one line and is ellipsized if too long to fit. The item will draw a background based on
  * the [isSelected] and [isActive] values.
  */
+@Deprecated("Use the overload with selected, active, colorFilter and hints")
+@ScheduledForRemoval(inVersion = "Before 2025.3")
```

Comment:
We know exactly when, don't be vague:

```suggestion
@ScheduledForRemoval(inVersion = "In 2025.3")
```

(same below)